modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/QuantumToolkit.java

```diff
@@ -1497,44 +1553,91 @@ public void run() {
+                        if (pImage.image == null) {
+                            pImage.setImage(com.sun.prism.Image.fromIntArgbPreData(IntBuffer.allocate(w * h), w, h));
+                        }
+                        // Find out if it is possible to divide up the image in tiles of the same size
+                        int tileWidth = computeTileSize(w, maxTextureSize);
+                        int tileHeight = computeTileSize(h, maxTextureSize);
+                        IntBuffer buffer = IntBuffer.allocate(tileWidth * tileHeight);
+
+                        // M represents the middle set of tiles each with a size of tileW x tileH.
+                        // R is the right hand column of tiles,
+                        // B is the bottom row,
+                        // C is the corner:
+                        // +-----------+-----------+  .  +-------+
+                        // |           |           |  .  |       |
+                        // |     M     |     M     |  .  |   R   |
+                        // |           |           |  .  |       |
+                        // +-----------+-----------+  .  +-------+
+                        // |           |           |  .  |       |
+                        // |     M     |     M     |  .  |   R   |
+                        // |           |           |  .  |       |
+                        // +-----------+-----------+  .  +-------+
+                        //       .           .        .      .
+                        // +-----------+-----------+  .  +-------+
+                        // |     B     |     B     |  .  |   C   |
+                        // +-----------+-----------+  .  +-------+
+
+                        // Walk through all same-size "M" tiles
+                        int xOffset = 0;
+                        int yOffset = 0;
+                        var mTileWidth = tileWidth;
+                        var mTileHeight = tileHeight;
+                        while (mTileWidth == tileWidth && xOffset < w) {
+                            yOffset = 0;
+                            mTileHeight = tileHeight;
+                            while (mTileHeight == tileHeight && yOffset < h) {
+                                renderTile(x, xOffset, y, yOffset, mTileWidth, mTileHeight,
+                                        buffer, rf, tileRttCache, pImage);
+                                yOffset += tileHeight;
+                                mTileHeight = Math.min(tileHeight, h - yOffset);
+                            }
+                            xOffset += tileWidth;
+                            mTileWidth = Math.min(tileWidth, w - xOffset);
+                        }
+                        // Walk through remaining same-width "B" tiles, if any
+                        int bOffset = 0;
+                        int bTileHeight = tileHeight;
+                        while (bTileHeight == tileHeight && bOffset < h) {
```

Comment:
It looks like the "B" and the "R" loops are reversed. This isn't causing any actual problems, but is confusing since it doesn't match the comment or the diagram above. The comment for this block says "B" tiles, but actually, it is the "R" tiles in the diagram that this is looping over. At the end of the main loop, `mTileWIdth` and `mTileHeight` will be set to the size of the corner tile. Given this, the tiles of `mTileWidth` X `tileHeight` will be the left hand column.

Once you fix this, you will need to surround this `while` loop in a check for `if (mTileWidth > 0)`

Reply:
You're right: the comments and variable names are swapped. Will fix it (and obviously add the extra condition that prevents attempts to render 0 width/height tiles). 
Thanks.